### DISCUSSION TITLE
Macro to shift Z plane down

### DISCUSSION THREAD

**Discussion by dastultz** [winder/Universal-G-Code-Sender#2581] [Help]
Hello, some years ago I had a macro that I've lost and I'm trying to recreate. Let's say I'm doing a face mill operation, cutting at Z0.0. I'm cleaning up rough cut stock. The operation finishes, but I didn't take enough off. I want to shift the whole operation down, say 10 mils, and run it again. 

I believe this can be done by clicking the current Z position and entering some math expression like -0.01 or something. I believe I was doing this in a macro. I see there is {machine_z} which is the current Z position, but I don't see any way to do a G10 operation where Z is set to the current Z minus some value. Can that still be done?

Issue #1426 is basically what I'm after. I can't remember when I had my macro working. I've been in and out of the hobby for years. If no math expressions can be done in a macro, I'm curious what others are using the substitutions {machine_z} and others for.

Is there some other way to do it? Currently I have to move the machine to Z-0.01, then hit the Z button to zero it out. I was achieving the same thing before but with no movement, much faster. Thanks.

**Comment by breiler** [comment]
The {machine_z} variable does not support math expressions yet. 

You could do this by clicking the Z position and enter `# + 0.01`, which will take the current position (#) and add 0.01.

Or you could do this with a offset command in a macro as well, but you need to be a Z0 when issuing it:
```
G10 P0 L20 Z0.01
```

Both of these will effectively "raise" your machine coordinate, so the next time you run your job it will run it 0.01 deeper.
